LPS-115810 - Disable svg4everybody for non-ie browsers and only if cdn is disabled

bryceosterhaus opened the pull request:
https://issues.liferay.com/browse/LPS-115810

Okay, so I'm not totally positive on some of these changes, but I was able to confirm that this does work. svg4everybody is only loaded if the browser is IE and if the cdn is enabled. I'll go through and make a few comments about areas I am unsure of.

liferay-continuous-integration commented:
To conserve resources, the PR Tester does not automatically run for every pull.

If your code changes were already tested in another pull, reference that pull in this pull so the test results can be analyzed.

If your pull was never tested, comment &quot;ci:test&quot; to run the PR Tester for this pull.

bryceosterhaus commented on modules/apps/adaptive-media/adaptive-media-web/src/main/resources/META-INF/resources/adaptive_media/js/utils/entryIcons.es.js:
```diff
@@ -15,19 +15,15 @@
 export function disableEntryIcon(element) {
 	element.parentElement.classList.add('disabled');
 
-	element.setAttribute('data-href', element.getAttribute('href'));
 	element.setAttribute('data-onclick', element.getAttribute('onclick'));
 
-	element.removeAttribute('href');
 	element.removeAttribute('onclick');
 }
 
 export function enableEntryIcon(element) {
 	element.parentElement.classList.remove('disabled');
 
-	element.setAttribute('href', element.getAttribute('data-href'));
 	element.setAttribute('onclick', element.getAttribute('data-onclick'));
 
-	element.removeAttribute('data-href');
```
Unsure of these changes, but since we don't intend to use `data-href` for icons, it seemed appropriate to remove this.

jbalsas replied:
Makes sense to me 👍

bryceosterhaus commented on modules/apps/frontend-js/frontend-js-svg4everybody-web/build.gradle:
```diff
@@ -0,0 +1,44 @@
+import com.liferay.gradle.util.copy.StripPathSegmentsAction
+
+configurations {
+	coreJSBundle
+}
```
Is this necessary? I copy/pasted it from the ie-compatability module

jbalsas replied:
No, this is unnecessary. This defines the line below as a configuration

> coreJSBundle group: "org.webjars.npm", name: "core-js-bundle", transitive: false, version: "3.1.4"

That means that we're downloading that thingie from maven and then you can use it.

In the IE module we download core-js that way and then extract its contents. Definitely not needed here.

bryceosterhaus commented on modules/apps/frontend-js/frontend-js-svg4everybody-web/src/main/java/com/liferay/frontend/js/svg4everybody/web/internal/servlet/taglib/SVG4EverybodyTopHeadDynamicInclude.java:
```diff
@@ -0,0 +1,97 @@
+			HttpServletResponse httpServletResponse, String key)
+		throws IOException {
+
+		if (!PropsValues.CDN_DYNAMIC_RESOURCES_ENABLED ||
```
I _think_ `CDN_DYNAMIC_RESOURCES_ENABLED ` is the right prop to check against, but I'm not totally positive. Should I check against `cdn.host.http` and `cdn.host.https` instead?

jbalsas replied:
Sounds reasonable

jbalsas commented:
This looks YUMMY!!!

![shipit](https://user-images.githubusercontent.com/905006/89591021-6eb83680-d849-11ea-8fb7-3b37bb5f1678.gif)

Nah, just kidding... we're going to wait until GA5 is out to push this forward, but then...

![shipit](https://user-images.githubusercontent.com/905006/89591040-78da3500-d849-11ea-9106-f313d7e2cb30.gif)

jbalsas commented on modules/apps/frontend-js/frontend-js-svg4everybody-web/build.gradle:
```diff
@@ -0,0 +1,44 @@
+	compileOnly group: "org.osgi", name: "osgi.core", version: "6.0.0"
+	compileOnly project(":apps:portal-url-builder:portal-url-builder-api")
+
+	coreJSBundle group: "org.webjars.npm", name: "core-js-bundle", transitive: false, version: "3.1.4"
```
This shouldn't be needed either

jbalsas commented on modules/apps/frontend-js/frontend-js-svg4everybody-web/package.json:
```diff
@@ -0,0 +1,12 @@
+{
+	"dependencies": {
+		"svg4everybody": "^2.1.9"
```
I doubt they'll be releasing new versions of `svg4everybody` but maybe pin this down?

jbalsas commented on util-taglib/src/com/liferay/taglib/aui/ATag.java:
```diff
@@ -60,7 +60,7 @@ protected int processEndTag() throws Exception {
 				jspWriter.write("<svg class=\"lexicon-icon ");
 				jspWriter.write("lexicon-icon-shortcut\" focusable=\"false\" ");
-				jspWriter.write("role=\"img\"><use data-href=\"");
+				jspWriter.write("role=\"img\"><use href=\"");
```
Loving this!

To make this foolproof, I was thinking that maybe this should be:

```
String hrefAttr = "href";

if (CDN) hrefAttr = "data-href"; // Note, no need to distinguish IE here

jspWriter.write(...);
```

That being said, the only effect of this in CDN would be that we'd see a rejected request to the icons spritemap. We probably can't control all calling sites anyways, with people generating raw markup like there's no tomorrow, so I'm leaning towards just keeping it simple... 

Thoughts?

jbalsas replied:
One offender would be `Clay`... and I doubt we want to go add this everywhere in Clay :D

bryceosterhaus replied:
@jbalsas so are you thinking we would still run svg4everybody with `attributeName: 'data-href',` at https://github.com/liferay-frontend/liferay-portal/pull/220/files#diff-541d787ebc908170523bf4b674fa67afL66?

If so, we would likely want to check IE as well since we run svg4everybody with at attr. Or the other option is to also check for cdn where we instantiate svg4everybody

jbalsas replied:
> @jbalsas so are you thinking we would still run svg4everybody with attributeName: 'data-href', at https://github.com/liferay-frontend/liferay-portal/pull/220/files#diff-541d787ebc908170523bf4b674fa67afL66?

Yes, but never mind that. To really pull that off, we should make sure there's no `use href` anywhere in the whole page or contributed html... which I doubt we want. The only possible way to do this would be using a `filter` to rewrite the whole page before serving it. But filters are expensive... so I don't think this is worth it.

Just remember this in case it comes up in the future as something we might need to do :)

bryceosterhaus replied:
Yeah thats what I was thinking, there are a handful of places that don't use the tag version and we would need it there as well. I'll add the other updates and try and measure some performance and then run tests to see where we are at with this.

jbalsas commented on util-taglib/src/com/liferay/taglib/aui/IconTag.java:
```diff
@@ -129,7 +129,7 @@ private void _processIconContent(PageContext pageContext) {
 				jspWriter.write(
 					InlineUtil.buildDynamicAttributes(getDynamicAttributes()));
-				jspWriter.write("><use data-href=\"");
+				jspWriter.write("><use href=\"");
```
Same would apply as in `ATag`

jbalsas replied:
Let's forget about this for now

bryceosterhaus commented:
Just updated and squashed the changes. One update I made was at https://github.com/liferay-frontend/liferay-portal/pull/220/files#diff-36f08efd8f1b93dfdb940957b7457352L83 which I'm not totally sure why we call svg4everybody again there.

I wasn't able to find any noticeable measurements via lighthouse, but here is the before/after of svg requests. The lighthouse measurements where all over the place from run to run, so I didn't really trust it.

Before:
![Screen Shot 2020-08-07 at 4 24 54 PM](https://user-images.githubusercontent.com/6843530/89696239-d0b98000-d8cb-11ea-9b94-d1d0970b9e0c.png)

After:
![Screen Shot 2020-08-07 at 4 22 16 PM](https://user-images.githubusercontent.com/6843530/89696241-d1eaad00-d8cb-11ea-9f71-76b3a266c4b5.png)


 So we at least mitigate a request that takes 311ms.

bryceosterhaus commented:
ci:test:sf

bryceosterhaus commented:
ci:test:relevant

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-2.liferay.com/job/test-portal-source-format/3475/">test-portal-source-format</a></p><p><b>For more details click <a href="https://test-1-2.liferay.com//userContent/jobs/test-portal-source-format/builds/3475/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest(master)/6903/">test-portal-acceptance-pullrequest(master)</a></p><p><b>For more details click <a href="https://test-1-2.liferay.com//userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/6903/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:sf - 1 out of 1 jobs passed in 3 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/275058a5906bcdc9a083795b9c98c2c7860a1995">275058a5906bcdc9a083795b9c98c2c7860a1995</a></p><h4>Sender Branch:</h4><p>Branch Name: <a href="https://github.com/bryceosterhaus/liferay-portal/tree/LPS-115810">LPS-115810</a><br/>Branch GIT ID: <a href="https://github.com/bryceosterhaus/liferay-portal/commit/dfe8bbe42b84f41359d33c24dcdbb6d274a561f5">dfe8bbe42b84f41359d33c24dcdbb6d274a561f5</a></p>1 out of 1jobs PASSED<details><summary><strong>1 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-2.liferay.com/job/test-portal-source-format/3475/">test-portal-source-format</a></li></ul></details><h5>For more details click <a href="https://test-1-2.liferay.com/userContent/jobs/test-portal-source-format/builds/3475/jenkins-report.html">here</a>.</h5></details></html>

liferay-continuous-integration commented:
<strong>Jenkins Build:</strong><a href="https://test-1-2.liferay.com/job/test-portal-source-format/3475/">test-portal-source-format#3475</a><br /><strong>Jenkins Report:</strong><a href="https://test-1-2.liferay.com/userContent/jobs/test-portal-source-format/builds/3475/jenkins-report.html">jenkins-report.html</a><br /><strong>Jenkins Suite:</strong> sf<br /><strong>Pull Request:</strong><a href="https://github.com/liferay-frontend/liferay-portal/pull/220">liferay-frontend#220</a><br /><strong>Spira Release:</strong><a href="https://liferay.spiraservice.net/16/Release/1302.aspx">/Liferay DXP 7.3/Pull Request/ci:test:sf</a><br /><strong>Spira Release Build:</strong><a href="https://liferay.spiraservice.net/16/Build/26075.aspx">master - bryceosterhaus > liferay-frontend - PR#220 - 2020-08-07[16:41:01]</a><br /><strong>Spira Jenkins Build:</strong><a href="https://test-1-20.liferay.com/job/publish-spira-report/2246/">publish-spira-report#2246</a><br />

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:stable - 21 out of 21 jobs passed</h3><h3>:heavy_check_mark: ci:test:relevant - 72 out of 73 jobs passed in 2 hours 27 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/275058a5906bcdc9a083795b9c98c2c7860a1995">275058a5906bcdc9a083795b9c98c2c7860a1995</a></p><h4>Copied in Private Modules Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal-ee/tree/master-private">master-private</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal-ee/commit/a4eb9471e502819c1cfc9d16ef0eea0078d40a10">a4eb9471e502819c1cfc9d16ef0eea0078d40a10</a></p><details><summary><strong>ci:test:stable - 21 out of 21 jobs PASSED</strong></summary><details><summary><strong>21 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/279181/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/450759/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-9.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/340911/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/1</a></li><li><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/341957/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/10</a></li><li><a href="https://test-1-21.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/369105/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/11</a></li><li><a href="https://test-1-12.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/310166/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/2</a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/418482/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/3</a></li><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/419838/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/4</a></li><li><a href="https://test-1-21.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/369104/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/5</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/390452/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/6</a></li><li><a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/279187/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/7</a></li><li><a href="https://test-1-18.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/180648/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/8</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/438629/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/9</a></li><li><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/425490/">test-portal-acceptance-pullrequest-batch(master)/js-unit-jdk8</a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/289844/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql57-jdk8_stable</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/438630/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-jdk8_stable</a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/382461/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-project-templates-jdk8</a></li><li><a href="https://test-1-12.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/310165/">test-portal-acceptance-pullrequest-batch(master)/pmd-jdk8</a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/418479/">test-portal-acceptance-pullrequest-batch(master)/rest-builder-jdk8</a></li><li><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/425489/">test-portal-acceptance-pullrequest-batch(master)/service-builder-jdk8</a></li><li><a href="https://test-1-7.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/399189/">test-portal-acceptance-pullrequest-batch(master)/unit-jdk8_stable</a></li></ul></details></details><details><summary><strong>ci:test:relevant - 71 out of 73 jobs PASSED</strong></summary><div><h4>2 Failed Jobs:</h4><ul><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest(master)/6903/"><strike><strong>test-portal-acceptance-pullrequest(master)</strong></strike></a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/289845/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/semantic-versioning-jdk8</strong></strike></a></li></ul></div><details><summary><strong>71 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-7.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/399188/">test-portal-acceptance-pullrequest-batch(master)/central-requirements-jdk8</a></li><li><a href="https://test-1-14.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/427906/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-mysql57-jdk11_open/0</a></li><li><a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/279181/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-7.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/399187/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/0</a></li><li><a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/279185/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/1</a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/418481/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/10</a></li><li><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/425488/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/11</a></li><li><a href="https://test-1-9.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/340910/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/12</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/438628/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/13</a></li><li><a href="https://test-1-12.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/310164/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/14</a></li><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/419836/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/15</a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/382459/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/16</a></li><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/419837/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/17</a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/382460/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/18</a></li><li><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/425443/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/19</a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/450757/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/2</a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/450755/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/20</a></li><li><a href="https://test-1-14.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/427903/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/21</a></li><li><a href="https://test-1-12.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/310162/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/22</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/438625/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/23</a></li><li><a href="https://test-1-14.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/427901/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/24</a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/450753/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/25</a></li><li><a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/279182/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/26</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/438623/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/27</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/438624/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/28</a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/418480/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/3</a></li><li><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/425442/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/4</a></li><li><a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/279180/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/5</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/438627/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/6</a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/450756/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/7</a></li><li><a href="https://test-1-14.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/427904/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/8</a></li><li><a href="https://test-1-12.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/310163/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/9</a></li><li><a href="https://test-1-9.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/340908/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/0</a></li><li><a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/279184/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/1</a></li><li><a href="https://test-1-14.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/427905/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/10</a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/450758/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/11</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/438626/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/2</a></li><li><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/425441/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/3</a></li><li><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/425487/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/4</a></li><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/419835/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/5</a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/382458/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/6</a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/450754/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/7</a></li><li><a href="https://test-1-14.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/427902/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/8</a></li><li><a href="https://test-1-12.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/310161/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/9</a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/450759/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-9.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/340911/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/1</a></li><li><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/341957/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/10</a></li><li><a href="https://test-1-21.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/369105/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/11</a></li><li><a href="https://test-1-12.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/310166/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/2</a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/418482/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/3</a></li><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/419838/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/4</a></li><li><a href="https://test-1-21.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/369104/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/5</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/390452/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/6</a></li><li><a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/279187/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/7</a></li><li><a href="https://test-1-18.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/180648/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/8</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/438629/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/9</a></li><li><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/425490/">test-portal-acceptance-pullrequest-batch(master)/js-unit-jdk8</a></li><li><a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/279183/">test-portal-acceptance-pullrequest-batch(master)/lpkg-base-jdk8</a></li><li><a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/279186/">test-portal-acceptance-pullrequest-batch(master)/lpkg-controller-jdk8/null</a></li><li><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/341958/">test-portal-acceptance-pullrequest-batch(master)/modules-compile-jdk8</a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/289843/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql57-jdk8</a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/289844/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql57-jdk8_stable</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/390451/">test-portal-acceptance-pullrequest-batch(master)/modules-semantic-versioning-jdk8</a></li><li><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/425444/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-jdk8</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/438630/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-jdk8_stable</a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/382461/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-project-templates-jdk8</a></li><li><a href="https://test-1-12.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/310165/">test-portal-acceptance-pullrequest-batch(master)/pmd-jdk8</a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/418479/">test-portal-acceptance-pullrequest-batch(master)/rest-builder-jdk8</a></li><li><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/425489/">test-portal-acceptance-pullrequest-batch(master)/service-builder-jdk8</a></li><li><a href="https://test-1-9.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/340909/">test-portal-acceptance-pullrequest-batch(master)/unit-jdk8</a></li><li><a href="https://test-1-7.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/399189/">test-portal-acceptance-pullrequest-batch(master)/unit-jdk8_stable</a></li></ul></details></details><h5>For more details click <a href="https://test-1-2.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/6903/jenkins-report.html">here</a>.</h5><hr/><h4>This pull contains no unique failures.</h4><hr/><details><summary><strong>Failures in common with <a href="https://test-1-1.liferay.com/job/test-portal-acceptance-upstream-dxp(master)">acceptance upstream results</a> at 275058a5906bcdc9a083795b9c98c2c7860a1995:</strong></summary><ol><li><div><h5><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/289845/">test-portal-acceptance-pullrequest-batch(master)/semantic-versioning-jdk8</a></h5><div><h6>Job Results:</h6><p>0 Tests Passed.<br/>1 Test Failed.</p></div><ol><li><div><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=0,label_exp=!master/289845//consoleText">AXIS_VARIABLE=0,label_exp=!master #289845</a><pre><code>   [delete] Deleting: /opt/dev/projects/github/liferay-portal/clean_up_java_processes.sh
[stopwatch] [run.batch.test.setup: 23:09.684 sec]
     [echo] run.batch.test.action.start.timestamp: 08-07-2020 18:22:20:895 PDT
     [echo] Checking for baseline log files.
     [echo] 
     [echo] ##
     [echo] ## /opt/dev/projects/github/liferay-portal/baseline-reports/portal-impl-5.log
     [echo] ##
     [echo] 
     [echo] [Baseline Warning] Bundle Version Change Recommended: 5.13.2
     [echo] 
     [echo] ##
     [echo] ## /opt/dev/projects/github/liferay-portal/baseline-reports/portal-kernel-9.log
     [echo] ##
     [echo] 
     [echo] [Baseline Warning] Bundle Version Change Recommended: 9.0.2
     [echo] 
     [echo] ##
     [echo] ## /opt/dev/projects/github/liferay-portal/baseline-reports/portal-test-8.log
     [echo] ##
     [echo] 
     [echo] [Baseline Warning] Bundle Version Change Recommended: 8.1.1
     [echo] 
     [echo] ##
     [echo] ## /opt/dev/projects/github/liferay-portal/baseline-reports/support-tomcat-4.log
     [echo] ##
     [echo] 
     [echo] [Baseline Warning] Bundle Version Change Recommended: 4.0.4
     [echo] 
     [echo] ##
     [echo] ## /opt/dev/projects/github/liferay-portal/baseline-reports/util-bridges-7.log
     [echo] ##
     [echo] 
     [echo] [Baseline Warning] Bundle Version Change Recommended: 7.0.5
     [echo] 
     [echo] ##
     [echo] ## /opt/dev/projects/github/liferay-portal/baseline-reports/util-java-5.log
     [echo] ##
     [echo] 
     [echo] [Baseline Warning] Bundle Version Change Recommended: 5.0.7
     [echo] 
     [echo] ##
     [echo] ## /opt/dev/projects/github/liferay-portal/baseline-reports/util-slf4j-5.log
     [echo] ##
     [echo] 
     [echo] [Baseline Warning] Bundle Version Change Recommended: 5.1.2
     [echo] 
     [echo] ##
     [echo] ## /opt/dev/projects/github/liferay-portal/baseline-reports/util-taglib-5.log
     [echo] ##
     [echo] 
     [echo] [Baseline Warning] Bundle Version Change Recommended: 5.2.1
[stopwatch] [run.batch.test.action: 9.290 sec]
     [echo] Semantic versioning is incorrect.
      [get] Getting: http://test-1-22/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=0,label_exp=!master/289845//consoleText
      [get] To: /opt/dev/projects/github/liferay-portal/20200807182230185.txt
   [delete] Deleting: /opt/dev/projects/github/liferay-portal/20200807182230185.txt
   [delete] Deleting: /opt/dev/projects/github/liferay-portal/null219293247.properties</code></pre></div></li></ol></div></li></ol></details></details></html>

liferay-continuous-integration commented:
<strong>Jenkins Build:</strong><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest(master)/6903/">test-portal-acceptance-pullrequest(master)#6903</a><br /><strong>Jenkins Report:</strong><a href="https://test-1-2.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/6903/jenkins-report.html">jenkins-report.html</a><br /><strong>Jenkins Suite:</strong> relevant<br /><strong>Pull Request:</strong><a href="https://github.com/liferay-frontend/liferay-portal/pull/220">liferay-frontend#220</a><br /><strong>Spira Release:</strong><a href="https://liferay.spiraservice.net/16/Release/1312.aspx">/Liferay DXP 7.3/Pull Request/ci:test:relevant</a><br /><strong>Spira Release Build:</strong><a href="https://liferay.spiraservice.net/16/Build/26106.aspx">master - bryceosterhaus > liferay-frontend - PR#220 - 2020-08-07[16:41:01]</a><br /><strong>Spira Jenkins Build:</strong><a href="https://test-1-16.liferay.com/job/publish-spira-report/5517/">publish-spira-report#5517</a><br />

wincent commented:
I know this is still only marked as "draft", but I also added the "on hold" label to make it clear that we don't want to merge it until GA5 is done and dusted. ([Via Slack](https://liferay.slack.com/archives/CLA7D1CNL/p1596816620245500), I get the impression that we didn't get a build yet.)

bryceosterhaus commented:
@wincent thanks! I am keeping an eye on whenever we cut a release and then I will move this to non-draft

jbalsas commented on modules/apps/frontend-js/frontend-js-svg4everybody-web/src/main/java/com/liferay/frontend/js/svg4everybody/web/internal/servlet/taglib/SVG4EverybodyTopHeadDynamicInclude.java:
```diff
@@ -0,0 +1,97 @@
+			HttpServletResponse httpServletResponse, String key)
+		throws IOException {
+
+		if (!PropsValues.CDN_DYNAMIC_RESOURCES_ENABLED ||
```
Shouldn't this be `if (PropsValues.CDN_DYNAMIC_RESOURCES_ENABLED)` instead of the negated version? 🤔

jbalsas replied:
Reading through the docs, it says:

```
    #
    # Set this to true to enable serving dynamically generated CSS, JavaScript,
    # and images via a CDN. Setting this to false allows the usage of CDNs that
    # do not support lazy loading of resources (e.g. Amazon CloudFront).
    #
    # Env: LIFERAY_CDN_PERIOD_DYNAMIC_PERIOD_RESOURCES_PERIOD_ENABLED
    #
    cdn.dynamic.resources.enabled=true
```

And we also have `LIFERAY_CDN_PERIOD_HOST_PERIOD_HTTP` and `LIFERAY_CDN_PERIOD_HOST_PERIOD_HTTPS`, so maybe we need a combination of these?

If I'm reading this correctly, one would need to set this propery to `false` now (which is different from the default) to benefit from this.

Maybe we want something like:

```java
PropsValues.CDN_DYNAMIC_RESOURCES_ENABLED && (PropsValues.LIFERAY_CDN_PERIOD_HOST_PERIOD_HTTP || PropsValues.LIFERAY_CDN_PERIOD_HOST_PERIOD_HTTPS)
```

jbalsas replied:
I see that based on https://issues.liferay.com/browse/LPS-71763, the steps to enable CDN are:

```
redirect.url.ips.allowed=
cdn.host.http=http://abc.com:8080/
cdn.dynamic.resources.enabled=false
```

So, even though I don't fully understand it, I'm going to forward so we can get eyes on this soon and then we can try to make sense of this :D

jbalsas commented:
Hey @bryceosterhaus, took this out of Draft and was about to forward, but had some doubts with when are we actually including svg4everybody. Could you take a look?

jbalsas commented:
Actually, see https://github.com/liferay-frontend/liferay-portal/pull/220#discussion_r471702503 :)

jbalsas commented:
ci:forward

liferay-continuous-integration commented:
CI is automatically triggering the following test suites:
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**relevant**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**sf**

The pull request will automatically be forwarded to the user `brianchandotcom` if the following test suites pass:
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**relevant**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**sf**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**stable**

liferay-continuous-integration commented:
Skipping previously passed test suites:
`ci:test:relevant`
`ci:test:sf`

liferay-continuous-integration commented:
All required test suite(s) passed.
Forwarding pullrequest to `brianchandotcom`.

liferay-continuous-integration commented:
Pull request has been successfully forwarded to https://github.com/brianchandotcom/liferay-portal/pull/92804
